### ISSUE TITLE
`profiles/hardened`: setting `environment.memoryAllocator.provider`to scudo occasionally breaks sshd 

### ISSUE THREAD

**Issue by al3xtjames** [NixOS/nixpkgs#157451]
### Describe the bug
After importing `profiles/hardened.nix`, I noticed that logging into my host via SSH would occasionally fail with one of the following errors:
```console
client_loop: send disconnect: Broken pipe
```

```console
Connection closed by <host IP> port 22
```

```console
Connection reset by <host IP> port 22
```

`journalctl -u sshd` on the remote host showed the following error:
```console
fatal: privsep_preauth: preauth child terminated by signal 31
```

I was able to reduce this to the allocator change. Setting `environment.memoryAllocator.provider = "scudo";` is enough to cause these occasional sshd crashes.

### Steps To Reproduce
1. Add  `environment.memoryAllocator.provider = "scudo";` to configuration.nix and switch to the new configuration on the remote host.
2. Try to repeatedly connect to the host with `while true; do ssh <host> true; done`.
3. Observe one of the error messages shown above.

### Expected behavior
SSH connections work as expected.

### Additional context
Here's the backtrace from a coredump:
```
(gdb) bt
#0  0x0000de92f957e828 in syscall () from /nix/store/5z7rlpz6dsnp5fgivz24kkc19va507h7-glibc-2.33-62/lib/libc.so.6
#1  0x0000de92f9b68554 in gwp_asan::AllocationMetadata::RecordAllocation(unsigned long, unsigned long) () from /nix/store/hkvgl5ka064a0c55vfwcc20li06jqg2s-malloc-provider-scudo/lib/libclang_rt.scudo-aarch64.so
#2  0x0000de92f9b69150 in gwp_asan::GuardedPoolAllocator::allocate(unsigned long, unsigned long) () from /nix/store/hkvgl5ka064a0c55vfwcc20li06jqg2s-malloc-provider-scudo/lib/libclang_rt.scudo-aarch64.so
#3  0x0000de92f9b56df0 in __scudo::Allocator::allocate(unsigned long, unsigned long, __scudo::AllocType, bool) () from /nix/store/hkvgl5ka064a0c55vfwcc20li06jqg2s-malloc-provider-scudo/lib/libclang_rt.scudo-aarch64.so
#4  0x0000de92f9b5337c in __scudo::scudoAllocate(unsigned long, unsigned long, __scudo::AllocType) () from /nix/store/hkvgl5ka064a0c55vfwcc20li06jqg2s-malloc-provider-scudo/lib/libclang_rt.scudo-aarch64.so
#5  0x0000de92f99d90ac in CRYPTO_zalloc () from /nix/store/qqk9kv2kma1iy9h8cipfhk8g5widswic-openssl-1.1.1m/lib/libcrypto.so.1.1
#6  0x0000de92f99c8530 in EVP_CipherInit_ex () from /nix/store/qqk9kv2kma1iy9h8cipfhk8g5widswic-openssl-1.1.1m/lib/libcrypto.so.1.1
#7  0x00000f5c1335a954 in chachapoly_new ()
#8  0x00000f5c1333eb48 in cipher_init ()
#9  0x00000f5c1334580c in ssh_set_newkeys ()
#10 0x00000f5c1335c110 in kex_input_newkeys ()
#11 0x00000f5c1334e3a4 in ssh_dispatch_run ()
#12 0x00000f5c1334e47c in ssh_dispatch_run_fatal ()
#13 0x00000f5c132f08a4 in main ()
```

And some output from strace:

```
[pid  3544] gettid()                    = ?
[pid  3544] +++ killed by SIGSYS (core dumped) +++
```

As shown in the backtrace, this `gettid` call is coming from [gwp_asan::AllocationMetadata::RecordAllocation](https://github.com/llvm/llvm-project/blob/llvmorg-13.0.0/compiler-rt/lib/gwp_asan/common.cpp#L46) ([`getThreadID` is a simple wrapper around `gettid`](https://github.com/llvm/llvm-project/blob/llvmorg-13.0.0/compiler-rt/lib/gwp_asan/platform_specific/common_posix.cpp#L20)).

`gettid` is not included in OpenSSH's seccomp whitelist: https://github.com/openssh/openssh-portable/blob/V_8_8_P1/sandbox-seccomp-filter.c#L133. So this should probably be fixed upstream.

### Notify maintainers
@joachifm
@emilazy

### Metadata
 - system: `"aarch64-linux"`
 - host os: `Linux 5.10.94, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"nixos-21.11.334797.6979c0e49bb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by al3xtjames** [comment]
This should be fixed by #161569.

**Comment by Artturin** [comment]
reopen if it happens again
